test/zdtm/static/socket_udp-corked.c

```diff
@@ -22,54 +23,101 @@ const char *test_author = "Pavel Emelyanov <xemul@parallels.com<>\n";
 #include <netinet/udp.h>
 
 static int port = 8880;
+static char buf[11];
 
 #define MSG1 "msg1"
+#define MSG2 "msg_2"
+#define MSG "msg1\0msg_2"
 
 int main(int argc, char **argv)
 {
-	int ret, sk1;
+	int ret, server_sk, client_sk;
 	socklen_t len = sizeof(struct sockaddr_in);
-	struct sockaddr_in addr1;
+	struct sockaddr_in server_addr, client_addr, addr;
 	int opt;
 
 	test_init(argc, argv);
 
-	sk1 = socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP);
-	if (sk1 < 0) {
+	server_sk = socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP); // server udp
+	if (server_sk < 0) {
 		pr_perror("Can't create socket");
 		return 1;
 	}
 
-	memset(&addr1, 0, sizeof(addr1));
-	addr1.sin_family = AF_INET;
-	addr1.sin_addr.s_addr = inet_addr("127.0.0.1");
-	addr1.sin_port = htons(port);
+	memset(&server_addr, 0, sizeof(server_addr));
+	server_addr.sin_family = AF_INET;
+	server_addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+	server_addr.sin_port = htons(port);
 
-	ret = bind(sk1, (struct sockaddr *)&addr1, len);
+	ret = bind(server_sk, (struct sockaddr *)&server_addr, len);
 	if (ret < 0) {
 		pr_perror("Can't bind socket");
 		return 1;
 	}
-	ret = connect(sk1, (struct sockaddr *)&addr1, len);
+
+	client_sk = socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP); // client udp
+	if (client_sk < 0) {
+		pr_perror("Can't create socket");
+		return 1;
+	}
+
+	memset(&client_addr, 0, sizeof(server_addr));
+	client_addr.sin_family = AF_INET;
+	client_addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+	client_addr.sin_port = htons(port + 1);
+
+	ret = bind(client_sk, (struct sockaddr *)&client_addr, len);
+	if (ret < 0) {
+		pr_perror("Can't bind socket");
+		return 1;
+	}
+	ret = connect(client_sk, (struct sockaddr *)&server_addr, len);
```

Comment:
&client_sk

Reply:
I don't think '&' is necessary here since connect() requires int value fd.

Reply:
I'm sorry, my bad. I intended to suggest the following change
```diff
- ret = connect(client_sk, (struct sockaddr *)&server_addr, len);
+ ret = connect(client_sk, (struct sockaddr *)&client_addr, len);
```

Reply:
You’re right